src/GraphQL/Language/AST/ValueNodes/ValueNode.cs

```diff
@@ -1,14 +1,35 @@
+#nullable enable
+
+using System;
+
 namespace GraphQL.Language.AST
 {
     /// <summary>
     /// Represents a value node which contains a literal value within a document.
     /// </summary>
     public abstract class ValueNode<T> : AbstractNode, IValue<T>
     {
+        [Obsolete]
+        public ValueNode() : this(default!)
+        {
+        }
+
+        public ValueNode(T value)
+        {
+#pragma warning disable CS0612 // Type or member is obsolete
+            Value = value;
+#pragma warning restore CS0612 // Type or member is obsolete
+        }
+
         /// <inheritdoc/>
-        public T Value { get; protected set; }
+        public T Value
+        {
+            get;
+            [Obsolete]
+            protected set;
+        }
 
-        object IValue.Value => Value;
+        object IValue.Value => Value!;
```

Comment:
`NullValue` returns `null`.

Reply:
In master I see ` object? Value { get; }` for `IValue` so need change here.

Reply:
That is because `NullNode` returns `null`.  `ValueNode<T>.Value` cannot return `null`.  So this code should be correct.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Language/AST/IValue.cs#L11

I see explicit `IValue` (not `IValue<T>`) interface implementation that returns `object` instead of `object?`. So was it planned?

Reply:
I think I see what you're saying.  Since it's an explicit `IValue` implementation, not exposed as a public property on the class, it might as well be `object? IValue.Value => Value;`.

That's true, it can be.  However, you could also say that this is more accurate source-code, depicting that a non-null value will always be returned, even if `IValue.Value` is defined as `object?`.  If you want to change it, that's fine; I don't care either way.

Reply:
Perhaps there should be a non-null type restriction added to the generic `T` type for v5.

Reply:
>  If you want to change it, that's fine; I don't care either way.

50/50, nevermind